src/classes/ADDR_OtherAcc_TEST.cls

```diff
@@ -0,0 +1,204 @@
+                                                                      UTIL_Describe.getBizAccRecTypeID() + ';'));
+ 
+        // create accounts without addresses
+        list<Account> listAcc = UTIL_UnitTestData_TEST.CreateMultipleTestAccounts(3, UTIL_Describe.getBizAccRecTypeID());
+        insert listAcc;
+
+         // create addresses for the accounts
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(listAcc.size());
+        for (Integer i = 0; i < listAcc.size(); i++) {
+            addrs[i].Parent_Account__c = listAcc[i].Id;
+            addrs[i].Default_Address__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        
+        // verify results
+        listAcc = [select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c 
+        from Account order by Account.Name];
+        system.assertEquals(3, listAcc.size());
+        
+        addrs = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, Parent_Account__c, 
+        Latest_Start_Date__c, Latest_End_Date__c from Address__c order by Parent_Account__r.Name];
+        system.assertEquals(3, addrs.size());
+        
+        for (Integer i = 0; i < listAcc.size(); i++) {
+            Account acc = listAcc[i];
+            Address__c addr = addrs[i];
+            system.assertEquals(acc.BillingStreet, addr.MailingStreet__c);
+            system.assertEquals(acc.BillingCity, addr.MailingCity__c);
+            system.assertNotEquals(null, acc.BillingStreet);
+            system.assertNotEquals(null, acc.BillingCity);
+            system.assertEquals(true, addr.Default_Address__c);
+            system.assertEquals(addr.Id, acc.Current_Address__c);
+        }
+    }
+    
+    /*********************************************************************************************************
+    @description Create Accounts when the setting for Address Mgmt is off. 
+    verify:
+        no Address objects created
+        Account addresses set
+    **********************************************************************************************************/            
+    @isTest
+    public static void testDisabledAddr() {        
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Accounts_Addresses_Enabled__c = null));
+
+        List<Account> accs = UTIL_UnitTestData_TEST.CreateMultipleTestAccounts(2, UTIL_Describe.getAdminAccRecTypeID());
+        for (Account acc : accs) {
+            acc.BillingStreet = 'new street';
+            acc.BillingCity = 'new city';
+        }
+
+        // verify Account address
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.